sopt/src/main/scala/dagr/sopt/ArgTokenizer.scala

```diff
@@ -69,19 +74,34 @@ class ArgTokenizer(args: TraversableOnce[String]) extends Iterator[Try[ArgTokeni
   }
 
   /** Returns any remaining args that were not tokenized. This is a destructive operation, so should only be called once. */
-  def takeRemaining: Traversable[String] = this.iterator.toTraversable
+  def takeRemaining: Traversable[String] = this.stack.toTraversable
 
   /** Sets the `nextToken` if it is not defined and we have more strings in the iterator */
-  private def updateNextToken(): Unit = {
-    if (nextToken.isEmpty && iterator.hasNext) {
-      nextToken = iterator.next() match {
-        case "--" => None
-        case ""   => Some(Failure(new OptionNameException("Empty argument given.")))
-        case arg if arg.startsWith("--") => Some(convertDoubleDashOption(arg.substring(2)))
-        case arg if arg.startsWith("-") => Some(convertSingleDash(arg.substring(1)))
-        case arg => Some(Success(ArgValue(value = arg)))
+  private def updateNextToken(): Unit = if (nextToken.isEmpty && this.stack.nonEmpty) nextToken = takeNextToken
+
+  /**
+    * Requires that the stack have at least one item left in it; pulls the top item from the
+    * stack and processes it.
+    */
+  private def takeNextToken: Option[Try[Token]] = (this.stack.pop(), argFilePrefix) match {
+    case ("--", _) => None
+    case ("",   _) => Some(Failure(new OptionNameException("Empty argument given.")))
+    case (arg,  _) if arg.startsWith("--") => Some(convertDoubleDashOption(arg.substring(2)))
+    case (arg,  _) if arg.startsWith("-") => Some(convertSingleDash(arg.substring(1)))
+    case (arg, Some(pre)) if arg.startsWith(pre) =>
+      try {
+        this.stack.pushAll(loadArgumentFile(arg.drop(pre.length)).reverseIterator)
+        if (this.stack.nonEmpty) takeNextToken else None
       }
-    }
+      catch {
+        case ex: IOException => Some(Failure(ex))
+      }
```

Comment:
This is the section I pointed out in the mean thread.  I tried all kinds of things here to use a `Try`, and got tied in knots.  What I wanted to do was have `loadArgumentFile` return a Try[List[String]] instead of throwing exception, but it's all complicated by the fact that
1. This method returns an `Option[Try[Token]]`
2. On line 94 I need to return a straight `None`  in the case where the arguments file is the last argument in the stack, and the arguments file turns out to be empty

It's beyond me (currently) to do this with `Try`, but I'd be glad of some help if you think it's doable.